server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -204,11 +196,9 @@ synchronized boolean hasSnapshottedCommits() {
     boolean hasUnreferencedCommits() throws IOException {
         final IndexCommit lastCommit = this.lastCommit;
         if (safeCommit != lastCommit) { // Race condition can happen but harmless
-            if (lastCommit.getUserData().containsKey(SequenceNumbers.MAX_SEQ_NO)) {
```

Comment:
👍 we assert this always happens in `IndexShard#assertSequenceNumbersInCommit()`.